tests/integration/utils2.go

```diff
@@ -1457,6 +1424,10 @@ func updateDoc(
 	}
 
 	assertExpectedErrorRaised(s.t, s.testCase.Description, action.ExpectedError, expectedErrorRaised)
+
+	if action.ExpectedError == "" {
+		waitForUpdateEvents(s, action.NodeID, action.CollectionID)
```

Comment:
questions: this looks like unsolicited sync. Why is it unconditional?

Reply:
There should always be an update event published to the local event bus if a document is updated or created.